fendermint/app/src/app.rs

```diff
@@ -108,6 +108,12 @@ impl AppState {
 
         tendermint::hash::AppHash::try_from(state_params_hash).expect("hash can be wrapped")
     }
+
+    /// The state is effective at the *next* block, that is, the effects of block N are visible in the header of block N+1,
```

Comment:
So essentially we are doing deferred execution. Block N points to the state of height N-1, right? We should document this clearly somewhere (not sure where).

Also, @cryptoAtwill, worth checking how the operation of top-down is impacted by this change (I guess there will be a minor impact because this is already how Lotus works, but we strongly rely on committing the block hash an state of a specific height, but we must ensure that we don't introduce any inconsistencies).